### PR TITLE
37-refactor-sequence-class

### PR DESCRIPTION
**Pull request by Milad-Aalto**
Edit files:
sequence.py
rl_grid_lin_curr_ctr.py
rl_grid_mpc_enum.py

### CODE REVIEW

File: soft4pes/utils/sequence.py [hunk 1]
```diff
@@ -59,11 +46,19 @@ def __call__(self, t):
...
         """
-        return np.array([
-            np.interp(t, self.times, self.values[:, 0], period=self._period),
-            np.interp(t, self.times, self.values[:, 1], period=self._period)
-        ])
+        # Define the dimensions of columns to iterate over
+        dim = range(self.values.shape[1])
+
+        #Initialize a list to hold the interpolated values
+        interpolated_values = []
```
Comment (artturuusila) [inline soft4pes/utils/sequence.py:57]: It is obvious what this line does, no need for the line comment

File: soft4pes/utils/sequence.py [hunk 2]
```diff
@@ -59,11 +46,19 @@ def __call__(self, t):
...
+
+        # Perform interpolation for each dimension
+        for m in dim:
```
Comment (artturuusila) [inline soft4pes/utils/sequence.py:60]: No need to make the additional dim value,` for m in range(self.values.shape[1]):` is clear and the line is not too long.

File: soft4pes/utils/sequence.py [hunk 3]
```diff
@@ -59,11 +46,19 @@ def __call__(self, t):
...
-            np.interp(t, self.times, self.values[:, 0], period=self._period),
-            np.interp(t, self.times, self.values[:, 1], period=self._period)
-        ])
+        # Define the dimensions of columns to iterate over
+        dim = range(self.values.shape[1])
+
+        #Initialize a list to hold the interpolated values
+        interpolated_values = []
+
+        # Perform interpolation for each dimension
```
Comment (artturuusila) [inline soft4pes/utils/sequence.py:59]: For each column. I may have used word dimension in the task description, but the correct term is column (we always have two dimensions)

**Review by artturuusila — APPROVED** [review]

**Comment by Milad-Aalto** [comment]
closes #37

**Pull request merged**